app/api/v_sphere/folder.rb

```diff
@@ -70,13 +70,17 @@ def move_here(folder_entry)
     end
 
     def create_vm(cpu, ram, capacity, name, cluster)
-      vm_config = creation_config(cpu, ram, capacity, name)
+      vm_config = creation_config(cpu, ram, capacity, prefix(name))
       vm = @folder.CreateVM_Task(config: vm_config, pool: cluster.resource_pool).wait_for_completion
       VSphere::VirtualMachine.new vm
     end
 
     private
 
+    def prefix(name)
+      "#{Time.now.strftime('%Y%m%d')}_vm-#{name}"
```

Comment:
This method returns the full name with prefix of a vm, right? `vm_name_with_prefix(name)`, `add_prefix(name)`, `format_vm_name(name)`?
Why is this in `folder.rb`?

Reply:
It is in the folder.rb because vms are created here
It is just a helper method

Reply:
> It is in the folder.rb because vms are created here

Let me rephrase my question: Is there a specific reason why methods for creating VMs (and associated helper methods) are in file called `folder.rb` and not in e.g. `app/api/v_sphere/virtual_machine.rb`? I would have expected VM creation to have to do more with VMs and less with folders.

Reply:
> Let me rephrase my question: Is there a specific reason why methods for creating VMs (and associated helper methods) are in file called folder.rb and not in e.g. app/api/v_sphere/virtual_machine.rb? I would have expected VM creation to have to do more with VMs and less with folders.

The VM creation is in folder.rb because the vSphere's API-call to create Virtual Machines is only available on folders. For the simple reason, that you always need a folder to create the vm in. We simply kept this convention.
And we kept the associated helper methods in the same file to make them easy to find, associate them with the action they are helping and because virtual_machine.rb is already quite large.

Reply:
> vSphere's API-call to create Virtual Machines is only available on folders
Ah, I see, that is the critical information. Thanks.